test/e2e_env/universal/observability/metrics.go

```diff
@@ -0,0 +1,110 @@
+package observability
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	. "github.com/kumahq/kuma/test/framework"
+	"github.com/kumahq/kuma/test/framework/client"
+	"github.com/kumahq/kuma/test/framework/envs/universal"
+)
+
+func MeshWithMetricsTLS(mesh string) InstallFunc {
+	yaml := fmt.Sprintf(`
+type: Mesh
+name: %s
+metrics:
+  enabledBackend: prometheus-1
+  backends:
+  - name: prometheus-1
+    type: prometheus
+    conf:
+      path: /metrics
+      port: 1234
+      tls:
+        enabled: true
+        mode: delegated`, mesh)
+	return YamlUniversal(yaml)
+}
+
+func PrometheusMetrics() {
+	const mesh = "prometheus-metrics"
+	const clientName = "demo-client-outside-mesh"
+
+	BeforeAll(func() {
+		err := NewClusterSetup().
+			Install(MeshWithMetricsTLS(mesh)).
+			Install(TestServerUniversal("test-server", mesh,
+				WithTransparentProxy(true),
+				WithArgs([]string{"echo", "--instance", "test-server"}),
+				WithServiceName("test-server"),
+				WithDpEnvs(map[string]string{
+					"KUMA_DATAPLANE_METRICS_CERT_PATH": "/kuma/server.crt",
+					"KUMA_DATAPLANE_METRICS_KEY_PATH":  "/kuma/server.key",
+				}),
+			)).
+			Install(TestServerUniversal("test-server-no-tls", mesh,
+				WithTransparentProxy(true),
+				WithArgs([]string{"echo", "--instance", "test-server-no-tls"}),
+				WithServiceName("test-server-no-tls"),
+			)).
+			Install(DemoClientUniversal(clientName, "", WithoutDataplane())).
+			Setup(universal.Cluster)
+		Expect(err).ToNot(HaveOccurred())
+	})
+	E2EAfterAll(func() {
+		Expect(universal.Cluster.DeleteMeshApps(mesh)).To(Succeed())
+		Expect(universal.Cluster.DeleteMesh(mesh)).To(Succeed())
+		Expect(universal.Cluster.DeleteApp(clientName)).To(Succeed())
+	})
+
+	It("should expose metrics through https", func() {
+		// given
+		host := universal.Cluster.GetApp("test-server").GetIP()
+
+		// when doing http request
+		_, _, err := client.CollectResponse(
+			universal.Cluster,
+			clientName,
+			"http://test-server.mesh:1234/metrics",
+			client.Resolve("test-server.mesh:1234", fmt.Sprintf("[%s]", host)),
+		)
```

Comment:
why is it possible to still do HTTP? 🤔

Reply:
Client can try to do the request but it fails with HTTP, I am just checking here if it fails `Expect(err).To(HaveOccurred())`

Reply:
ah, I see